modules/util/path.go

```diff
@@ -173,3 +177,22 @@ func FileURLToPath(u *url.URL) (string, error) {
 	}
 	return path, nil
 }
+
+// HomeDir returns path of '~'(in Linux) on Windows,
+// it returns error when the variable does not exist.
+func HomeDir() (home string, err error) {
+	if isOSWindows() {
+		home = os.Getenv("USERPROFILE")
+		if home == "" {
+			home = os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
+		}
+	} else {
+		home = os.Getenv("HOME")
+	}
+
+	if home == "" {
+		return "", errors.New("cannot get home directory")
+	}
+
+	return home, nil
```

Comment:
Could we use something along the lines of:

```go
    user, err := user.Current()
    if err != nil {
        return err
    }
    return user.HomeDir
```

Reply:
Not sure, it's breaking. I believe some user run Gitea with "HOME=xxx" and mismatched uid ......

Reply:
A new comment added:
```
	// TODO: some users run Gitea with mismatched uid  and "HOME=xxx" (they set HOME=xxx by environment manually)
	// so at the moment we can not use `user.Current().HomeDir`
```